File: core/src/com/symbol/game/ecs/Mapper.java
package com.symbol.game.ecs;

import com.badlogic.ashley.core.ComponentMapper;
import com.symbol.game.ecs.component.*;
import com.symbol.game.ecs.component.enemy.*;
import com.symbol.game.ecs.component.map.*;

public class Mapper {

    public static final ComponentMapper<PositionComponent> POS_MAPPER = ComponentMapper.getFor(PositionComponent.class);
    public static final ComponentMapper<VelocityComponent> VEL_MAPPER = ComponentMapper.getFor(VelocityComponent.class);
    public static final ComponentMapper<TextureComponent> TEXTURE_MAPPER = ComponentMapper.getFor(TextureComponent.class);
    public static final ComponentMapper<BoundingBoxComponent> BOUNDING_BOX_MAPPER = ComponentMapper.getFor(BoundingBoxComponent.class);
    public static final ComponentMapper<GravityComponent> GRAVITY_MAPPER = ComponentMapper.getFor(GravityComponent.class);
    public static final ComponentMapper<JumpComponent> JUMP_MAPPER = ComponentMapper.getFor(JumpComponent.class);
    public static final ComponentMapper<DirectionComponent> DIR_MAPPER = ComponentMapper.getFor(DirectionComponent.class);
    public static final ComponentMapper<PlayerComponent> PLAYER_MAPPER = ComponentMapper.getFor(PlayerComponent.class);
    public static final ComponentMapper<RemoveComponent> REMOVE_MAPPER = ComponentMapper.getFor(RemoveComponent.class);
    public static final ComponentMapper<ProjectileComponent> PROJ_MAPPER = ComponentMapper.getFor(ProjectileComponent.class);
    public static final ComponentMapper<HealthComponent> HEALTH_MAPPER = ComponentMapper.getFor(HealthComponent.class);
    public static final ComponentMapper<KnockbackComponent> KNOCKBACK_MAPPER = ComponentMapper.getFor(KnockbackComponent.class);
    public static final ComponentMapper<OrbitComponent> ORBIT_MAPPER = ComponentMapper.getFor(OrbitComponent.class);

    public static final ComponentMapper<EnemyComponent> ENEMY_MAPPER = ComponentMapper.getFor(EnemyComponent.class);
    public static final ComponentMapper<ActivationComponent> ACTIVATION_MAPPER = ComponentMapper.getFor(ActivationComponent.class);
    public static final ComponentMapper<CorporealComponent> CORPOREAL_MAPPER = ComponentMapper.getFor(CorporealComponent.class);
    public static final ComponentMapper<AttackComponent> ATTACK_MAPPER = ComponentMapper.getFor(AttackComponent.class);
    public static final ComponentMapper<ExplodeComponent> EXPLODE_MAPPER = ComponentMapper.getFor(ExplodeComponent.class);
    public static final ComponentMapper<TeleportComponent> TELEPORT_MAPPER = ComponentMapper.getFor(TeleportComponent.class);
    public static final ComponentMapper<LastStandComponent> LAST_STAND_MAPPER = ComponentMapper.getFor(LastStandComponent.class);
    public static final ComponentMapper<TrapComponent> TRAP_MAPPER = ComponentMapper.getFor(TrapComponent.class);

    public static final ComponentMapper<MapEntityComponent> MAP_ENTITY_MAPPER = ComponentMapper.getFor(MapEntityComponent.class);
    public static final ComponentMapper<MovingPlatformComponent> MOVING_PLATFORM_MAPPER = ComponentMapper.getFor(MovingPlatformComponent.class);
    public static final ComponentMapper<PortalComponent> PORTAL_MAPPER = ComponentMapper.getFor(PortalComponent.class);
    public static final ComponentMapper<ClampComponent> CLAMP_MAPPER = ComponentMapper.getFor(ClampComponent.class);
    public static final ComponentMapper<HealthPackComponent> HEALTH_PACK_MAPPER = ComponentMapper.getFor(HealthPackComponent.class);
    public static final ComponentMapper<ColorComponent> COLOR_MAPPER = ComponentMapper.getFor(ColorComponent.class);
    public static final ComponentMapper<SquareSwitchComponent> SQUARE_SWITCH_MAPPER = ComponentMapper.getFor(SquareSwitchComponent.class);
    public static final ComponentMapper<ToggleTileComponent> TOGGLE_TILE_MAPPER = ComponentMapper.getFor(ToggleTileComponent.class);

}

File: core/src/com/symbol/game/ecs/Mapper.kt
package com.symbol.game.ecs

import com.badlogic.ashley.core.ComponentMapper
import com.symbol.game.ecs.component.*
import com.symbol.game.ecs.component.enemy.*
import com.symbol.game.ecs.component.map.*

object Mapper {

    val POS_MAPPER = ComponentMapper.getFor(PositionComponent::class.java)!!
    val VEL_MAPPER = ComponentMapper.getFor(VelocityComponent::class.java)!!
    val TEXTURE_MAPPER = ComponentMapper.getFor(TextureComponent::class.java)!!
    val BOUNDING_BOX_MAPPER = ComponentMapper.getFor(BoundingBoxComponent::class.java)!!
    val BOUNDING_CIRCLE_MAPPER = ComponentMapper.getFor(BoundingCircleComponent::class.java)!!
    val GRAVITY_MAPPER = ComponentMapper.getFor(GravityComponent::class.java)!!
    val JUMP_MAPPER = ComponentMapper.getFor(JumpComponent::class.java)!!
    val DIR_MAPPER = ComponentMapper.getFor(DirectionComponent::class.java)!!
    val PLAYER_MAPPER = ComponentMapper.getFor(PlayerComponent::class.java)!!
    val REMOVE_MAPPER = ComponentMapper.getFor(RemoveComponent::class.java)!!
    val PROJ_MAPPER = ComponentMapper.getFor(ProjectileComponent::class.java)!!
    val HEALTH_MAPPER = ComponentMapper.getFor(HealthComponent::class.java)!!
    val KNOCKBACK_MAPPER = ComponentMapper.getFor(KnockbackComponent::class.java)!!
    val ORBIT_MAPPER = ComponentMapper.getFor(OrbitComponent::class.java)!!
    val STATUS_EFFECT_MAPPER = ComponentMapper.getFor(StatusEffectComponent::class.java)!!
    val AFFECT_ALL_MAPPER = ComponentMapper.getFor(AffectAllComponent::class.java)!!
    val LAST_ENTITY_MAPPER = ComponentMapper.getFor(LastEntityComponent::class.java)!!

    val ENEMY_MAPPER = ComponentMapper.getFor(EnemyComponent::class.java)!!
    val ACTIVATION_MAPPER = ComponentMapper.getFor(ActivationComponent::class.java)!!
    val CORPOREAL_MAPPER = ComponentMapper.getFor(CorporealComponent::class.java)!!
    val ATTACK_MAPPER = ComponentMapper.getFor(AttackComponent::class.java)!!
    val EXPLODE_MAPPER = ComponentMapper.getFor(ExplodeComponent::class.java)!!
    val TELEPORT_MAPPER = ComponentMapper.getFor(TeleportComponent::class.java)!!
    val LAST_STAND_MAPPER = ComponentMapper.getFor(LastStandComponent::class.java)!!
    val TRAP_MAPPER = ComponentMapper.getFor(TrapComponent::class.java)!!
    val BLOCK_MAPPER = ComponentMapper.getFor(BlockComponent::class.java)!!

    val MAP_ENTITY_MAPPER = ComponentMapper.getFor(MapEntityComponent::class.java)!!
    val MOVING_PLATFORM_MAPPER = ComponentMapper.getFor(MovingPlatformComponent::class.java)!!
    val PORTAL_MAPPER = ComponentMapper.getFor(PortalComponent::class.java)!!
    val CLAMP_MAPPER = ComponentMapper.getFor(ClampComponent::class.java)!!
    val HEALTH_PACK_MAPPER = ComponentMapper.getFor(HealthPackComponent::class.java)!!
    val COLOR_MAPPER = ComponentMapper.getFor(ColorComponent::class.java)!!
    val SQUARE_SWITCH_MAPPER = ComponentMapper.getFor(SquareSwitchComponent::class.java)!!
    val TOGGLE_TILE_MAPPER = ComponentMapper.getFor(ToggleTileComponent::class.java)!!
    val FORCE_FIELD_MAPPER = ComponentMapper.getFor(ForceFieldComponent::class.java)!!
    val DAMAGE_BOOST_MAPPER = ComponentMapper.getFor(DamageBoostComponent::class.java)!!
    val MIRROR_MAPPER = ComponentMapper.getFor(MirrorComponent::class.java)!!
    val INVERT_SWITCH_MAPPER = ComponentMapper.getFor(InvertSwitchComponent::class.java)!!

}